components/server/src/user/env-var-service.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { UserDB } from "@gitpod/gitpod-db/lib";
+import { UserEnvVar, UserEnvVarValue } from "@gitpod/gitpod-protocol";
+import { inject, injectable } from "inversify";
+import { Authorizer } from "../authorization/authorizer";
+import { IAnalyticsWriter } from "@gitpod/gitpod-protocol/lib/analytics";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { v4 as uuidv4 } from "uuid";
+import { Config } from "../config";
+
+@injectable()
+export class EnvVarService {
+    constructor(
+        @inject(Config) private readonly config: Config,
+        @inject(UserDB) private readonly userDB: UserDB,
+        @inject(Authorizer) private readonly auth: Authorizer,
+        @inject(IAnalyticsWriter) private readonly analytics: IAnalyticsWriter,
+    ) {}
+
+    async getAllEnvVars(requestorId: string, userId: string): Promise<UserEnvVarValue[]> {
+        await this.auth.checkPermissionOnUser(requestorId, "read_info", userId);
+        const result: UserEnvVarValue[] = [];
+        for (const value of await this.userDB.getEnvVars(userId)) {
+            result.push({
+                id: value.id,
+                name: value.name,
+                value: value.value,
+                repositoryPattern: value.repositoryPattern,
+            });
+        }
+        return result;
+    }
+
+    async setEnvVar(requestorId: string, userId: string, variable: UserEnvVarValue): Promise<void> {
+        await this.auth.checkPermissionOnUser(requestorId, "write_info", userId);
```

Comment:
I find it a bit confusing that we re-use `write-info`. I understand that we are re-using the same relationship, and we maybe don't want to have a permission for every API call. But `info` sounds too much like "metadata (potentially everybody can read)" to me, while env vars are sometime used to store secrets.

Reply:
Ok, I have updated the desciption in the issue accordingly.
> For permission checks we use `read_env_var` of the project resp. user resources and `write_env_var` when we write.